templates/node/node.py

```diff
@@ -0,0 +1,277 @@
+        instance = vdc.ovc.instance
+        self._ovc = j.clients.openvcloud.get(instance=instance)
+
+        return self._ovc
+
+    @property
+    def vdc(self):
+        if self._vdc:
+            return self._vdc
+
+        # Get object for an VDC service, make sure exactly one is running
+        vdc = self.api.services.find(template_uid=self.VDC_TEMPLATE, name=self.data['vdc'])
+        if len(vdc) != 1:
+            raise RuntimeError('found %s vdc, requires exactly 1' % len(vdc))
+        
+        self._vdc = vdc[0]
+
+        return self._vdc
+
+    @property
+    def space(self):
+        return self.ovc.space_get(
+            accountName=None,
+            spaceName=self.data['vdc']
+            )
+
+    @property
+    def sshkey(self):
+        """ Get a path and keyname of the sshkey service """
+
+        sshkeys = self.api.services.find(template_uid=self.SSH_TEMPLATE)
+        if len(sshkeys) == 0:
+            raise RuntimeError('no %s ssh services found' % len(sshkeys))
+
+        # Get key name and path
+        path = sshkeys[0].data['path']
+        key = path.split('/')[-1]
+
+        return key         
+
+    @property
+    def machine(self):
+        if self._machine:
+            return self._machine
+        return self.space.machines.get(self.name)
+
+    def install(self):
+        try:
+            self.state.check('actions', 'install', 'ok')
+            return
+        except StateCheckError:
+            pass        
+
+        machine = self.machine
+        if not machine:
+            machine = self._machine_create()
+
+        # Get data from the vm
+        ip_private, vm_info = machine.machineip_get()
+        self.data['sshLogin'] = vm_info['accounts'][0]['login']
+        self.data['sshPassword']= vm_info['accounts'][0]['password']
+        self.data['ipPrivate'] = ip_private
+        self.data['ipPublic'] = machine.space.model['publicipaddress']
+        self.data['machineId'] = machine.id
+
+        self.portforward_create(self.data['ports'])
+        self._configure_disks()
+        self.save()
+
+        self.state.set('actions', 'install', 'ok')
+
+    def _machine_create(self):
+        """ Create a new machine """
+        data = self.data
+        space = self.space
+        self._machine = space.machine_create(
+            name=self.name,
+            sshkeyname= self.sshkey,
+            image=data['osImage'],
+            disksize=data['bootDiskSize'],
+            datadisks=[data['dataDiskSize']],
+            sizeId=data['sizeId'],
+            )
+
+        return self._machine
+
+    def _configure_disks(self):
+        """
+        Configure one boot disk and one data disk when installing a machine.
+        """
+        machine = self.machine
+        space_name = machine.space.model['name']
+        if machine:
+            machine.start()
+        else:
+            raise RuntimeError('machine %s in not found' % self.name)        
+
+        for disk in machine.disks:
```

Comment:
I really don't believe we need to model the boot and data disk of the machine, it's much easier and cleaner to assume the machine (including its boot and data disk) as a single entity, that are operated at one (for example, destroy) 

This also can cause confusion once we start creating persistent disks and move them around machines. Hence, i strongly recommend we don't `create` a disk service here.

@FastGeert and @chrisvdg can u confirm, or give us input on this?